gcc/rust/rust-session-manager.cc

```diff
@@ -474,6 +498,26 @@ Session::enable_dump (std::string arg)
 void
 Session::parse_files (int num_files, const char **files)
 {
+  rust_assert (num_files > 0);
+
+  if (options.crate_name.empty ())
+    {
+      /* HACK: We use the first file to infer the crate name, which might be
+       * incorrect: since rustc only allows one file to be supplied in the
+       * command-line */
+      auto crate_name = infer_crate_name (files[0]);
+      rust_debug_loc (Location (), "inferred crate name: %s",
+		      crate_name.c_str ());
+      if (!options.set_crate_name (crate_name))
+	{
+	  rust_error_at (
+	    Location (),
```

Comment:
I actually want to fake location information here so that it points to the `<input-file>:1` or something like that.

So that `(inferred from the input file %<%s%>)` could be omitted.